src/main/java/com/appium/schema/CapabilitySchemaValidator.java

```diff
@@ -59,4 +64,24 @@ private void isPlatformInEnv() {
         }
     }
 
+    private void validateRemoteHosts(Schema schema){
+        CapabilityManager capabilityManager;
+        try {
+            capabilityManager = CapabilityManager.getInstance();
+            if (capabilityManager.getCapabilities().has("hostMachines")) {
+                JSONArray hostMachines = capabilityManager.getHostMachineObject();
+                for (Object hostMachine : hostMachines) {
+                    JSONObject hostMachineJson = ((JSONObject) hostMachine);
+                    String machineIP = (String) hostMachineJson.get("machineIP");
+                    if (InetAddress.getByName(machineIP).isReachable(5000)){
+                        System.out.println("ATD is Running on " + machineIP);
+                    } else {
+                        figlet("Unable to connect to Remote Host " + machineIP);
```

Comment:
Cool, I like the approach. Would recommend throwing an exception here.

Reply:
Added exception and removed unused argument